README.md

```diff
@@ -224,10 +224,12 @@ bandwidth. To avoid this, you can use the "partial clone" feature by replacing
 the steps with this:
 
 ```yaml
+      - name: Configure Git
+      - run: git config --global http.https://github.com/.extraheader "Authorization: Basic $(echo -n x-access-token:${{ github.token }} | base64 --wrap=0)"
       - name: Partial clone
-        run: git clone --bare --depth=1 --single-branch --filter=blob:none ${{ github.event.repository.html_url }} .
-      - name: Configure push token
-        run: git config http.https://github.com/.extraheader "Authorization: Basic $(echo -n x-access-token:${{ github.token }} | base64 --wrap=0)"
+        env:
+          ref: ${{ github.event.ref }}
+        run: git clone --bare --depth=100 --single-branch --branch ${ref#refs/heads/} --filter=blob:none ${{ github.event.repository.html_url }} .
```

Comment:
```suggestion
        run: git clone --bare --depth=100 --single-branch --branch ${ref#refs/heads/} --filter=blob:none ${{ github.event.repository.clone_url }} .
```

Not directly related top this change, but since last time I noticed there is `clone_url` (which includes the `.git` suffix) that felt slightly more appropriate for what we want.